packages/gatsby-theme-store/src/gatsby-ssr.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import type { RenderBodyArgs } from 'gatsby'
+
+import { handleUserBackOnline } from './sdk/offline'
+
+export const onRenderBody = ({
+  setPostBodyComponents,
+  pathname,
+}: RenderBodyArgs) => {
+  if (!pathname.startsWith('/offline')) {
+    return null
+  }
+
+  // The following `<script>` tag should only be added to an offline page.
+  return setPostBodyComponents([
+    <script
```

Comment:
Yep, because we don't always have React. When you try to navigate to a page which Gatsby hasn't already cached the resources, it will not run `hydrate()` on it, so you can only serve static HTML.

In the console, it outputs an error message such as

```
Uncaught (in promise) Error: page resources for /fashion-eyeglasses/p not found. Not rendering React.
    at production-app.js:164
```

That's why I went with adding this `<script>` tag. Because this will work in all cases, whether we have React or not.